File: packages/ui-swiftui/Sources/KinetixUI/Calendar.swift
//
// Calendar.swift — KinetixCalendar.
//
// Mirrors packages/ui/src/components/calendar.tsx (a react-day-picker
// month grid). SwiftUI has no standalone calendar view, so this wraps
// `DatePicker(.graphical)` — a month grid with selection — themed with
// `.tint`, the same "reuse the platform's widget, restyle it" call the
// Compose port made wrapping Material3's `DatePicker`. Single-date only;
// no range select (a documented gap, same as the Compose port).
//

import SwiftUI

public struct KinetixCalendar: View {
    @Environment(\.kinetixColors) private var colors

    @Binding private var selection: Date
    private let range: ClosedRange<Date>?

    public init(selection: Binding<Date>, in range: ClosedRange<Date>? = nil) {
        self._selection = selection
        self.range = range
    }

    public var body: some View {
        Group {
            if let range {
                DatePicker("", selection: $selection, in: range, displayedComponents: [.date])
            } else {
                DatePicker("", selection: $selection, displayedComponents: [.date])
            }
        }
        .datePickerStyle(.graphical)
        .labelsHidden()
        .tint(colors.primary)
    }
}

File: packages/ui-compose/ui/src/main/kotlin/com/kinetixui/ui/Calendar.kt
@file:OptIn(ExperimentalMaterial3Api::class)

package com.kinetixui.ui

import androidx.compose.material3.DatePicker
import androidx.compose.material3.DatePickerDefaults
import androidx.compose.material3.DatePickerState
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.rememberDatePickerState
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier

/**
 * KinetixCalendar — mirrors `packages/ui/src/components/calendar.tsx`
 * (a themed `react-day-picker` `DayPicker`). Compose's own Material3
 * `DatePicker` composable (`@ExperimentalMaterial3Api` at this project's
 * Material3 version — stable behavior, an opt-in) already renders exactly
 * that: a month grid with prev/next nav, day selection, today marker,
 * outside-day dimming — so this wraps it and only re-maps the colors onto
 * the KinetixUI token contract, the same "reuse the platform's widget,
 * restyle it" call `KinetixSlider`/`KinetixToaster`/`KinetixCarousel` made.
 * Range selection (`mode="range"` in the source) isn't exposed here —
 * Material3's own `DateRangePicker` would be a separate wrapper.
 */
@Composable
fun KinetixCalendar(
    state: DatePickerState = rememberDatePickerState(),
    modifier: Modifier = Modifier,
    showModeToggle: Boolean = false,
) {
    val colors = KinetixColorScheme.current
    DatePicker(
        state = state,
        modifier = modifier,
        showModeToggle = showModeToggle,
        colors = DatePickerDefaults.colors(
            containerColor = colors.background,
            titleContentColor = colors.mutedForeground,
            headlineContentColor = colors.foreground,
            weekdayContentColor = colors.mutedForeground,
            subheadContentColor = colors.mutedForeground,
            yearContentColor = colors.foreground,
            currentYearContentColor = colors.primary,
            selectedYearContentColor = colors.primaryForeground,
            selectedYearContainerColor = colors.primary,
            dayContentColor = colors.foreground,
            disabledDayContentColor = colors.mutedForeground,
            selectedDayContentColor = colors.primaryForeground,
            selectedDayContainerColor = colors.primary,
            todayContentColor = colors.primary,
            todayDateBorderColor = colors.primary,
        ),
    )
}
